File: yh_connect/scripts/yh_connect_sub_pub.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

import rospy
from std_msgs.msg import Int32,Float32
#두개인 이유 받고돌려보내야하므로

class ConnectSubPub:
    def __init__(self):
        self.sub = rospy.Subscriber("yh_connect_int",Int32,self.msgCallback)
        self.pub=  rospy.Publisher("yh_connect_float",Float32)
        self.float_msg=Float32()

    def msgCallback(self,msg):
        if (msg.data %5 ==0):
            self.float_msg.data=msg.data/3
            self.pub.publish(self.float_msg)


if __name__=="__main__":
    rospy.init_node("yh_connect_sub_pub")
    connect_sub_pub=ConnectSubPub()
    rospy.spin()

File: yh_connect/src/yh_connect_sub_pub.cpp
#include"ros/ros.h"
#include"std_msgs/Int32.h"
#include"std_msgs/Float32.h"

ros::Publisher pub;


void msgCallback(const std_msgs::Int32::ConstPtr& msg)
{
    if(msg->data %5 == 0)
    {
    std_msgs::Float32 float_msg;
    float_msg.data=(float)msg->data/3.0f;
    pub.publish(float_msg);
    }

}

int main(int argc,char** argv)
{
    ros::init(argc,argv,"yh_connect_sub_pub");
    ros::NodeHandle nh;

    ros::Subscriber sub=nh.subscribe("yh_connect_int",10,msgCallback);
    pub=nh.advertise<std_msgs::Float32>("yh_connect_float",10,msgCallback);

    ros::spin();

    return 0;

}
